### DISCUSSION TITLE
vectorization tool contour doesn't consider Linewidth

### DISCUSSION THREAD

**Discussion by Ratofl** [arkypita/LaserGRBL#1421] [Q&A]
When you use the vectorization tool the generated contourline is centerd around the original contour of the picture. however, to get the small details right (I use it to draw pcb Layouts) you need to shift the center of the contour line inwards for half the linewidth (for 0.3mm linewidth, which would be 3.33 lines/mm, that would be 0.15mm contourshift. Or for 0.1mm linewidth, it would be 10 lines/mm and 0.05 mm contourshift)

**Hardware and software configuration (please complete the following information):**
 - Grbl version 0.9h
 - LaserGRBL 4.2.1

Edit: just found a similar setting over at [lightburn](https://lightburnsoftware.github.io/NewDocs/CutSettings-Line.html#kerf-offset
)

**Comment by arkypita** [comment]
I don't understand. Can you add some pictures?

**Comment by Ratofl** [comment]
sure. snapshot of a part of some primitive pcb:
![img_485309_w](https://user-images.githubusercontent.com/80839819/112145495-bb832e80-8bda-11eb-8ac3-acedd99181de.png)
...vectorization...
![Unbenannt](https://user-images.githubusercontent.com/80839819/112294766-77079980-8c93-11eb-86ce-a839de46d28e.PNG)

...toolpath...
![Unbenannt2](https://user-images.githubusercontent.com/80839819/112145574-d190ef00-8bda-11eb-80e9-e24e72616f4d.PNG)

...problem...
![Unbenannt2 - Kopie](https://user-images.githubusercontent.com/80839819/112145599-da81c080-8bda-11eb-9d5a-1f0ea2b30702.PNG)

...real example of a print:
![IMG_20210323_210230_2](https://user-images.githubusercontent.com/80839819/112304393-ba670580-8c9d-11eb-999f-aff93939b2db.jpg)

...microscopic picture, bad practice vs good practice:
![LP0](https://user-images.githubusercontent.com/80839819/112451500-dded8700-8d55-11eb-8776-e9f0bb11645a.jpg)
![LP1](https://user-images.githubusercontent.com/80839819/112451504-de861d80-8d55-11eb-803a-bcb540b4004b.jpg)

I hope this helps. This is simply to demonstrate the problems with the contourline. I'm super content with the inset filling.
